src/util/gateway/index.ts

```diff
@@ -0,0 +1,21 @@
+export class GatewayError extends Error {}
+
+export enum GatewayPacketType {
+	Identify,
+	Hello
+}
+
+export interface GatewayPacket {
+	t: GatewayPacketType;
+}
+
+export interface IdentifyGatewayPacket extends GatewayPacket {
+	t: GatewayPacketType.Identify;
+	d: {
+		token: string;
+	};
+}
+
+export interface HelloGatewayPacket extends GatewayPacket {
+	t: GatewayPacketType.Hello;
+}
```

Comment:
Is it possible to have more descriptive names for the properties in the interface?

Reply:
Done!